x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/hooks/use_latest_function_config.ts

```diff
@@ -56,7 +56,8 @@ function getOptions(
     }));
 
   const sortFieldOptions: Array<EuiComboBoxOptionOption<string>> = indexPattern.fields
-    .filter((v) => !ignoreFieldNames.has(v.name) && v.sortable)
+    // The backend API for `latest` allows all field types for sort but the UI will be limited to `date`.
+    .filter((v) => !ignoreFieldNames.has(v.name) && v.sortable && v.type === 'date')
```

Comment:
Any ideas what we should do if there are no `date` fields in the source index?

![image](https://user-images.githubusercontent.com/7405507/104947723-18ccea80-59b4-11eb-8cef-ee8a2fc96492.png)

Reply:
An alternative solution could be to still make all fields available but group them in the dropdown by field type, with `date` fields first, then `numbers`, then `keyword`.

Something like

```
date fields

  @timestamp
  created_at

----
number fields

  responsetime

----
other

  airline
```

Screenshot of the grouped dropdown from EUI docs:

<img src="https://user-images.githubusercontent.com/230104/104949349-1a4fe000-59bf-11eb-90a9-5269a3c6b7e9.png" width="50%" />

Reply:
As discussed, let's stick to the current implementation and only show `date` type fields in the dropdown. Advanced users can switch to the API or dev tools if they really want to use a `number` field say. And add some help text below the Sort field input, along the lines of `Select the date field to be used to identify the latest document.`

Reply:
That text sounds fine to me.  I wonder if we can alternatively change the "Add a sort field...." text in the screenshot to "Add a date field ...." and maybe even "...that identifies the latest documents" if space allows.